libraries/core_libs/consensus/src/transaction_manager/transaction_manager.cpp

```diff
@@ -229,6 +230,27 @@ size_t TransactionManager::getNonfinalizedTrxSize() const {
   return nonfinalized_transactions_in_dag_.size();
 }
 
+std::vector<std::shared_ptr<Transaction>> TransactionManager::getNonfinalizedTrx(const std::vector<trx_hash_t> &hashes,
+                                                                                 bool sorted) {
+  std::vector<std::shared_ptr<Transaction>> ret;
+  ret.reserve(hashes.size());
+  std::shared_lock transactions_lock(transactions_mutex_);
+  for (const auto &hash : hashes) {
+    assert(nonfinalized_transactions_in_dag_.contains(hash));
+    ret.push_back(nonfinalized_transactions_in_dag_[hash]);
+  }
+  if (sorted) {
+    std::stable_sort(ret.begin(), ret.end(), [](const auto &t1, const auto &t2) {
+      if (t1->getSender() == t2->getSender()) {
+        return t1->getNonce() < t2->getNonce() ||
+               (t1->getNonce() == t2->getNonce() && t1->getGasPrice() > t2->getGasPrice());
+      }
+      return true;
+    });
```

Comment:
I remember you have a similar transactions sorting method in another place in another PR. I think that's better to make the transactions sorting to a unique method. If people change the transactions sorting in a later time, that will  be very possible only change in one place and forget another one.

Reply:
they are not exactly same and the other could change anytime (we could improve something), but this one is unchangeable as this is requirement to have correctly ordered transaction, where gasprice is important only if the nonce is same. For transaction dag ordering some validators, could include transaction with expensive gas, as they want to earn more on rewards